src/index.js

```diff
@@ -1,16 +1,11 @@
 function createActionEnhancerMiddleware(mappings) {
   return ({getState}) => next => action => {
     const enhancedFields = mappings.reduce((current, nextMapping) => {
-      let result = current;
-      if (action[nextMapping.id]) {
-        result = Object.assign({}, result);
-
-        if (nextMapping.mapState) {
```

Comment:
there was no point in checking mapState. it's the only possible thing to add. if it's not there, it's a mistake